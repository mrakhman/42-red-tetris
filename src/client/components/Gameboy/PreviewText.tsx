import React from 'react';
import { Text, View } from 'react-native';
import { styles } from '/client/components/Gameboy';

type Props = {
  isMultiplayerMode: boolean,
  opponentsNumber: number,
  isLeader?: boolean,
  gameover?: boolean,
}

const PreviewText = (props: Props): JSX.Element  => {
  const { isMultiplayerMode, opponentsNumber, isLeader, gameover } = props;
  const secondLineText = (opponentsNumber: number, isLeader?: boolean) => {
    if (opponentsNumber <= 0) return 'Wait for other players';
    if (opponentsNumber > 0 && isLeader) return 'Press Start to begin';
    if (opponentsNumber > 0 && !isLeader) return 'Wait for leader to start the game';
  };

  if (gameover) {
    return (
      <View style={{ alignItems: 'center' }}>
        <Text style={styles.gameover}>Game Over</Text>
        <Text style={styles.gameMode}>Please wait for other players to finish</Text>
      </View>
    );
  }
  if (isMultiplayerMode) {
    return (
      <View style={{ alignItems: 'center' }}>
        <Text style={styles.gameMode}>You are in multiplayer mode</Text>
        <Text style={styles.gameMode}>{opponentsNumber} other player(s) in your room</Text>
        <Text>{secondLineText(opponentsNumber, isLeader)}</Text>
      </View>
    );
  }
  return (
    <View style={{ alignItems: 'center' }}>
      <Text style={styles.gameMode}>You are in solo mode</Text>
      <Text>Press Play(P) to start</Text>
    </View>
  );
};

export default PreviewText;
