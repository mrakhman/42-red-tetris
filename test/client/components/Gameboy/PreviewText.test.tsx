import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PreviewText from '/client/components/Gameboy/PreviewText';

describe('PreviewText', () => {
  it('should render PreviewText for solo mode', () => {
    const props = {
      isMultiplayerMode: false,
      opponentsNumber: 0,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for Leader in multiplayer mode and no opponents', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 0,
      isLeader: true,
      gameover: false,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for not Leader in multiplayer mode and no opponents', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 0,
      isLeader: false,
      gameover: false,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for Leader in multiplayer mode with opponents', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 2,
      isLeader: true,
      gameover: false,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for not Leader in multiplayer mode with opponents', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 2,
      isLeader: false,
      gameover: false,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for Leader in multiplayer mode after gameover', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 1,
      isLeader: true,
      gameover: true,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PreviewText for not Leader in multiplayer mode after gameover', () => {
    const props = {
      isMultiplayerMode: true,
      opponentsNumber: 1,
      isLeader: false,
      gameover: true,
    };
    const wrapper = shallow(<PreviewText {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
