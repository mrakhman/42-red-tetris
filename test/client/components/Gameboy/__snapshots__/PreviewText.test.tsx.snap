// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewText should render PreviewText for Leader in multiplayer mode after gameover 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#990011",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Game Over
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    Please wait for other players to finish
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for Leader in multiplayer mode and no opponents 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    0
     other player(s) in your room
  </Text>
  <Text>
    Wait for other players
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for Leader in multiplayer mode with opponents < maxPlayersLimit 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    2
     other player(s) in your room
  </Text>
  <Text>
    Press Start to begin
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for Leader in multiplayer mode with opponents >= maxPlayersLimit 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    5
     other player(s) in your room
  </Text>
  <Text>
    Room is full, press Start to begin
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for not Leader in multiplayer mode after gameover 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#990011",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Game Over
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    Please wait for other players to finish
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for not Leader in multiplayer mode and no opponents 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    0
     other player(s) in your room
  </Text>
  <Text>
    Wait for other players
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for not Leader in multiplayer mode with opponents < maxPlayersLimit 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    2
     other player(s) in your room
  </Text>
  <Text>
    Wait for leader to start the game
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for not Leader in multiplayer mode with opponents >= maxPlayersLimit 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in multiplayer mode
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    5
     other player(s) in your room
  </Text>
  <Text>
    Room is full, ask leader to start the game
  </Text>
</View>
`;

exports[`PreviewText should render PreviewText for solo mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    You are in solo mode
  </Text>
  <Text>
    Press Play(P) to start
  </Text>
</View>
`;
